src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -217,9 +224,16 @@ private sealed class InProgressState : WithCompilationState
             /// </summary>
             private sealed class AllSyntaxTreesParsedState(
                 Compilation compilationWithoutGeneratedDocuments,
-                CompilationTrackerGeneratorInfo generatorInfo)
-                : WithCompilationState(compilationWithoutGeneratedDocuments, generatorInfo)
+                CompilationTrackerGeneratorInfo generatorInfo,
+                Compilation? compilationWithGeneratedDocuments) : WithCompilationTrackerState(compilationWithoutGeneratedDocuments, generatorInfo)
             {
+                /// <summary>
+                /// The result of taking the original completed compilation that had generated documents and updating them by
+                /// apply the <see cref="CompilationAndGeneratorDriverTranslationAction" />; this is not a correct snapshot in that
+                /// the generators have not been rerun, but may be reusable if the generators are later found to give the
+                /// same output.
+                /// </summary>
+                public Compilation? CompilationWithGeneratedDocuments { get; } = compilationWithGeneratedDocuments;
```

Comment:
previously, we would have code that would make an AllSyntaxTreesParsedState and then return its data, while also needing to return this *additional* piece of data, which would then move into the next state-machine transition.  By adding this info into AllSyntaxTreesParsedState itself, we cleanly can represent things as just state machine transitions.